californium-core/src/main/java/org/eclipse/californium/core/server/ServerMessageDeliverer.java

```diff
@@ -144,8 +179,8 @@ private void checkForObserveOption(final Exchange exchange, final Resource resou
 	 * @param list the path as list of resource names
 	 * @return the resource or null if not found
 	 */
-	private Resource findResource(final List<String> list) {
-		LinkedList<String> path = new LinkedList<String>(list);
+	protected final Resource findResource(final List<String> list) {
+		Deque<String> path = new LinkedList<String>(list);
```

Comment:
Why do you use the base `Deque` instead of the `LinkedList`? Seems to be only locally used, so I don't see the real benefit.

Reply:
FMPOV this is just a matter of style. We usually use the abstract (interface) type for declaring variables instead of using the concrete type.